### PR TITLE
Set `requires_grad` explicitly in `gradient_check` and function test

### PR DESCRIPTION
**Pull request by niboshi**

### CODE REVIEW

**Comment by hvy** [comment]
Does this change assume that the overridden forward won't call backward? It seems like batch renormalization is such a case and that it fails, or at least on this line https://github.com/chainer/chainer/blob/master/tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py#L145.

**Comment by niboshi** [comment]
Fixed.

**Comment by niboshi** [comment]
This PR is required by #6373.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
CI failures seem unrelated and LGTM!
